wandb/sdk/service/service_sock.py

```diff
@@ -0,0 +1,64 @@
+"""grpc service.
+
+Reliably launch and connect to grpc process.
+"""
+
+import logging
+from typing import TYPE_CHECKING
+
+from wandb.proto import wandb_server_pb2 as spb
+
+from .service_base import _pbmap_apply_dict
+from .service_base import ServiceInterface
+from ..lib.sock_client import SockClient
+
+if TYPE_CHECKING:
+    from wandb.sdk.wandb_settings import Settings
+
+
+class ServiceSockInterface(ServiceInterface):
+    _sock_client: SockClient
+
+    def __init__(self) -> None:
+        self._sock_client = SockClient()
+
+    def get_transport(self) -> str:
+        return "tcp"
+
+    def _get_sock_client(self) -> SockClient:
+        # TODO: remove this
```

Comment:
is this TODO need to be addressed?

Reply:
I removed the TODO, it just felt a bit ugly how backend/backend.py reaches into the implementation for grpc and socket.  It is not ideal, but it isnt much worse than other leaky abstractions we have currently.  I think it will be more obvious what to do here in the future.